File: Algorithms/C++/883-Projection_Area_of_3D_Shapes.cpp
// Runtime: 4 ms, faster than 98.88% of C++ online submissions for Projection Area of 3D Shapes.
// Memory Usage: 9.1 MB, less than 100.00% of C++ online submissions for Projection Area of 3D Shapes.
class Solution {
public:
    int projectionArea(vector<vector<int>>& grid) {
        int area = 0;
        for (vector<int>& i : grid) // over look
            for (int& j : i)
                area += j != 0;
        for (vector<int>& i : grid) // left look
            area += *max_element(i.begin(), i.end());
        for (int col = 0; col < grid.size(); ++col) {
            int high = 0;
            for (int row = 0; row < grid.size(); ++row)
                high = max(high, grid[row][col]);
            area += high;
        }
        return area;
    }
};

File: Algorithms/Python3.x/883-Projection_Area_of_3D_Shapes.py
# Runtime: 88 ms, faster than 56.80% of Python3 online submissions for Projection Area of 3D Shapes.
# Memory Usage: 13.8 MB, less than 12.50% of Python3 online submissions for Projection Area of 3D Shapes.
class Solution:
    def projectionArea(self, grid: List[List[int]]) -> int:
        area = 0
        for i in grid:  # over look
            for j in i:
                area += j != 0
        area += sum([max(s) for s in grid])  # left look
        for col in range(len(grid)):  # main look
            high = 0
            for row in range(len(grid)):
                high = max(high, grid[row][col])
            area += high
        return area
